sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_properties.asciidoc

```diff
@@ -0,0 +1,173 @@
+= sycl_ext_oneapi_usm_properties
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Introduction
+IMPORTANT: This specification is a draft.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+This extension introduces an alternative way to pass both runtime and compile time properties into USM malloc APIs.
+
+`malloc_device`, `malloc_host`, `malloc_shared` take the properties, pass runtime properties onto runtime libraries and create annotated_ptr base on compile-time properties.
```

Comment:
I wouldn't mention that run time properties are passed to the runtime; these are literally runtime APIs so that's self evident.  Returning an annotated_ptr is a change in API though relative to the other malloc overloads and is the vehicle through which this information makes it to the compiler so it makes sense to mention it.

Reply:
Thanks for the suggestions! now phrased as:

`malloc_device`, `malloc_host`, `malloc_shared` take the properties, and return an annotated_ptr that carries the passed in compile-time properties.